### PR TITLE
Feat/web farm task support

### PR DESCRIPTION
**Pull request by bkapustik**
# Motivation

* Adds support for vertical auto-scaling using web farm tasks.
* Fixes issue with invalid indexing results on multiple instances.

### CODE REVIEW

File: src/Kentico.Xperience.Lucene.Core/Scaling/IndexLogReusableItemWebFarmTask.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    public IndexLogReusableItemWebFarmTask()
+    {
+        eventLog = Service.Resolve<IEventLogService>();
+        luceneTaskLogger = Service.Resolve<ILuceneTaskLogger>();
+    }
+
+    public override void ExecuteTask()
+    {
+        string message = $"Server {SystemContext.ServerName} is processing a Lucene indexing task from creator {CreatorName}";
+        eventLog.LogInformation("Lucene Indexing Task", "Execute", message);
```
Comment (seangwright) [inline src/Kentico.Xperience.Lucene.Core/Scaling/IndexLogReusableItemWebFarmTask.cs:25]: How often will this log message appear in the Event Log?

File: src/Kentico.Xperience.Lucene.Core/Scaling/RebuildWebFarmTask.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public override async void ExecuteTask()
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+    {
+        string message = $"Server {SystemContext.ServerName} is processing a Lucene Rebuild task from creator {CreatorName}";
+        eventLog.LogInformation("Lucene Rebuild Task", "Execute", message);
+
+        luceneClient.Rebuild(IndexName!, default).GetAwaiter().GetResult();
```
Comment (seangwright) [inline src/Kentico.Xperience.Lucene.Core/Scaling/RebuildWebFarmTask.cs:27]:
```suggestion
    public override async Task ExecuteTask()
    {
        string message = $"Server {SystemContext.ServerName} is processing a Lucene Rebuild task from creator {CreatorName}";
        eventLog.LogInformation("Lucene Rebuild Task", "Execute", message);

       await luceneClient.Rebuild(IndexName!, default);
```

I don't think `async void` is a good choice. Either the method should be synchronous and call `.GetAwaiter().GetResult()` on the `Task` that `Rebuild()` returns or it should be asynchronous and return a `Task` to the caller.

File: src/Kentico.Xperience.Lucene.Core/Scaling/IndexLogReusableItemWebFarmTask.cs (hunk 1 above)
Comment (bkapustik) [reply src/Kentico.Xperience.Lucene.Core/Scaling/IndexLogReusableItemWebFarmTask.cs:25]: Fair point

File: src/Kentico.Xperience.Lucene.Core/Scaling/RebuildWebFarmTask.cs (hunk 2 above)
Comment (bkapustik) [reply src/Kentico.Xperience.Lucene.Core/Scaling/RebuildWebFarmTask.cs]: I deleted the `async void`. The `WebFarmTaskBase` does not provide a method with `Task` return type.

File: docs/Auto-Scaling.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Auto-scaling Support
+
+This integration supports Xperience's [Auto-scaling support](https://docs.kentico.com/developers-and-admins/configuration/auto-scaling-support). This means that your application can distribute computing among multiple applications and therefore increase the number of requests the project can handle.
```
Comment (DavidBKentico) [inline docs/Auto-Scaling.md:3]: The doc updates are ok, but let's use a permanent link here for the Auto-scaling support page:
https://docs.kentico.com/x/SI3WCQ

**Review by DavidBKentico — CHANGES_REQUESTED** [review]

**Review by DavidBKentico — APPROVED** [review]

**Review by michalJakubis — APPROVED** [review]

**Pull request merged**